cryodrgn/protocols/protocol_base.py

```diff
@@ -187,12 +189,22 @@ def createOutputStep(self):
         """ Creating a set of particles with z_values. """
         inputSet = self._getInputParticles()
         zValues = iter(self._getParticlesZvalues())
-        outImgSet = self._createSetOfParticles()
+        outImgSet = self._createSetOfParticlesFlex(progName=const.CRYODRGN)
         outImgSet.copyInfo(inputSet)
-        outImgSet.copyItems(inputSet, updateItemCallback=self._setZValues,
-                            itemDataIterator=zValues)
-        setattr(outImgSet, WEIGHTS, pwobj.String(self._getFileName('weights_final')))
-        setattr(outImgSet, CONFIG, pwobj.String(self._getFileName('config')))
+        outImgSet.setHasCTF(inputSet.hasCTF())
+        outImgSet.getFlexInfo().setProgName(const.CRYODRGN)
+
+        for particle, zValue in zip(inputSet.iterItems(), zValues):
```

Comment:
should we remove .iterItems() in this case? inputSet is already iterable

Reply:
Fewer characters and more concise

Reply:
`.iterItems()` removed :)